src/services/web3/web3.plugin.ts

```diff
@@ -0,0 +1,148 @@
+import { MetamaskConnector } from './connectors/metamask/metamask.connector';
+import { Connector } from './connectors/connector';
+import { computed, reactive, ref, Ref, toRefs } from 'vue';
+import { WalletConnectConnector } from './connectors/trustwallet/walletconnect.connector';
+import { getAddress } from '@ethersproject/address';
+import { lsGet, lsRemove, lsSet } from '@/lib/utils';
+import { Web3Provider } from '@ethersproject/providers';
+import { WalletLinkConnector } from './connectors/walletlink/walletlink.connector';
+
+export type Wallet = 'metamask' | 'walletconnect' | 'walletlink';
+export const SupportedWallets = [
+  'metamask',
+  'walletconnect',
+  'walletlink'
+] as Wallet[];
+export const WalletNameDictionary: Record<Wallet, string> = {
+  metamask: 'Metamask',
+  walletconnect: 'WalletConnect',
+  walletlink: 'Coinbase'
+};
+type ConnectorImplementation = new (...args: any[]) => Connector;
+export const Web3ProviderSymbol = Symbol('WEB3_PROVIDER');
+
+export type Web3Plugin = {
+  connectWallet: (wallet: Wallet) => Promise<void>;
+  disconnectWallet: () => Promise<void>;
+  provider: Ref<Web3Provider>;
+  account: Ref<string>;
+  chainId: Ref<number>;
+  connector: Ref<Connector>;
+  walletState: Ref<WalletState>;
+};
+
+const WalletConnectorDictionary: Record<Wallet, ConnectorImplementation> = {
+  metamask: MetamaskConnector,
+  walletconnect: WalletConnectConnector,
+  walletlink: WalletLinkConnector
+};
+
+type WalletState = 'connecting' | 'connected' | 'disconnected';
+type PluginState = {
+  connector: any;
+  walletState: WalletState;
+};
+
+export default {
+  install: async app => {
+    const alreadyConnectedAccount = ref(lsGet('connectedWallet', null));
+    const alreadyConnectedProvider = ref(lsGet('connectedProvider', null));
```

Comment:
Nice! So this allows us to not have to reload the connection all the time right?

Reply:
That is correct, it's mainly for when a user comes back to the app, it will load the provider + account that was previously used.